var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -236,15 +236,19 @@ def mplib_content(spec, pre=None):
     else:
         pre = mpi_spec.prefix
 
+    libname = 'mpi'
+    if 'mpich' in mpi_spec.name:
+        libname = 'mpich'
+
     info = {
         'name':   '{0}-{1}'.format(mpi_spec.name, mpi_spec.version),
         'prefix':  pre,
         'include': inc,
         'bindir':  bin,
         'libdir':  lib,
-        'FLAGS':  '-DOMPI_SKIP_MPICXX -DMPICH_IGNORE_CXX_SEEK',
+        'FLAGS':  '-DOMPI_SKIP_MPICXX -DMPICH_SKIP_MPICXX',
         'PINC':   '-I{0}'.format(inc),
-        'PLIBS':  '-L{0} -lmpi'.format(lib),
+        'PLIBS':  '-L{0} -l{1}'.format(lib, libname),
```

Comment:
That's a good idea, I didn't know about the `ld_flags`.  There's also a `mpi_spec.headers.cpp_flags` that can be used for `PINC`.  They both work fine for me when I try to compile.

So this:
```
'PINC':   '-I{0}'.format(inc),
'PLIBS':  '-L{0} -l{1}'.format(lib, libname)
```
can be replaced with this:
```
'PINC': mpi_spec.headers.cpp_flags,
'PLIBS': mpi_spec.libs.ld_flags,
```